src/libcore/iter.rs

```diff
@@ -2205,6 +2297,35 @@ impl<I> Iterator for Fuse<I> where I: Iterator {
     }
 
     #[inline]
+    fn nth(&mut self, n: usize) -> Option<I::Item> {
+        if self.done {
+            None
+        } else {
+            let nth = self.iter.nth(n);
```

Comment:
This only retains the semantics of `fuse` if we assume that `nth(n)` returns `None` whenever `n+1` calls to `next` would produce at least one `None`. That makes me slightly uneasy...